### ISSUE TITLE
Flask endpoint names may differ from function name

### ISSUE THREAD

**Issue by RazerM** [rycus86/prometheus_flask_exporter#23]
Thanks for fixing #22! It fixes the simple case but I also realised something else.

The problem with the following code is that Flask endpoint names can be different than the function name, consider `@app.route('/hello', endpoint='my_hello')`.

https://github.com/rycus86/prometheus_flask_exporter/blob/0203d62506a1119fa011c1b11036db95902d44c6/prometheus_flask_exporter/__init__.py#L465-L472

I think there is a nice solution though! See [Flask.view_functions](http://flask.pocoo.org/docs/1.0/api/?highlight=view_functions#flask.Flask.view_functions). This snippet corresponds to the quoted lines above:

```python
        if request.endpoint is not None:
            view_func = current_app.view_functions[request.endpoint]

            # There may be decorators 'above' us, but before the function is
            # registered with Flask
            while view_func != func:
                try:
                    view_func = view_func.__wrapped__
                except AttributeError:
                    break

            if view_func == func:
                # we are in a request handler method
                response_for_metric = make_response(response)
```

Note that func is the function itself:

https://github.com/rycus86/prometheus_flask_exporter/blob/0203d62506a1119fa011c1b11036db95902d44c6/prometheus_flask_exporter/__init__.py#L442-L443

**Comment by rycus86** [comment]
Oh, nice one, thanks a lot! Let me write a test for it and try your solution. Looking good from what I can tell. :)

**Comment by rycus86** [comment]
It worked perfectly, thank you very much again. It's running through CI now, should be soon available as `0.7.4` :tada:
